log/slog/slog.go

```diff
@@ -0,0 +1,172 @@
+}
+
+func (l *loggerWrapper) Debug(msg string, args ...any) {
+    level.Debug(log.With(defaultLoggerWrapper.logger, "msg", msg)).Log(args...)
+}
+
+func (l *loggerWrapper) Error(err error, msg string, args ...any) {
+	if err == nil {
+		level.Error(log.With(l.logger, "msg", msg)).Log(args...)
+		return
+	}
+
+	if msg == "" {
+		level.Error(log.With(l.logger, "err", err)).Log(args...)
+		return
+	}
+
+    level.Error(log.With(l.logger, "msg", msg, "err", err)).Log(args...)
+}
+
+func (l *loggerWrapper) Infof(format string, args ...any) {
+    level.Info(l.logger).Log("msg", fmt.Sprintf(format, args...))
+}
+
+func (l *loggerWrapper) Warnf(format string, args ...any) {
+    level.Warn(l.logger).Log("msg", fmt.Sprintf(format, args...))
+}
+
+func (l *loggerWrapper) Debugf(format string, args ...any) {
+    level.Debug(l.logger).Log("msg", fmt.Sprintf(format, args...))
+}
+
+func (l *loggerWrapper) Errorf(err error, format string, args ...any) {
+    level.Error(l.logger).Log("err", err.Error(), "msg", fmt.Sprintf(format, args...))
+}
+
+func (l *loggerWrapper) WithTraceId(ctx context.Context) *loggerWrapper {
+    traceId := instruments.ExtractTraceID(ctx)
+    logger := log.With(l.logger, "trace_id", traceId)
+
+	return &loggerWrapper{
+		logger: logger,
+	}
+}
+
+func (l *loggerWrapper) WithFields(fields map[string]any) *loggerWrapper {
+    fieldArgs := mapToSlice(fields)
+    logger := log.With(l.logger, fieldArgs...)
+
+	return &loggerWrapper{
+		logger: logger,
+	}
+}
+
+func Info(msg string, args ...any) {
+    level.Info(log.With(defaultLoggerWrapper.logger, "msg", msg)).Log(args...)
+}
+
+func Warn(msg string, args ...any) {
+    level.Warn(log.With(defaultLoggerWrapper.logger, "msg", msg)).Log(args...)
+}
+
+func Debug(msg string, args ...any) {
+    level.Debug(log.With(defaultLoggerWrapper.logger, "msg", msg)).Log(args...)
+}
+
+func Error(err error, msg string, args ...any) {
+	if err == nil {
+		level.Error(log.With(defaultLoggerWrapper.logger, "msg", msg)).Log(args...)
+		return
+	}
+
+	if msg == "" {
+		level.Error(log.With(defaultLoggerWrapper.logger, "err", err)).Log(args...)
+		return
+	}
+
+    level.Error(log.With(defaultLoggerWrapper.logger, "msg", msg, "err", err)).Log(args...)
+}
+
+func Infof(format string, args ...any) {
+    level.Info(defaultLoggerWrapper.logger).Log("msg", fmt.Sprintf(format, args...))
+}
+
+func Warnf(format string, args ...any) {
+    level.Warn(defaultLoggerWrapper.logger).Log("msg", fmt.Sprintf(format, args...))
+}
+
+func Debugf(format string, args ...any) {
+    level.Debug(defaultLoggerWrapper.logger).Log("msg", fmt.Sprintf(format, args...))
+}
+
+func Errorf(err error, format string, args ...any) {
+    level.Error(defaultLoggerWrapper.logger).Log("err", err.Error(), "msg", fmt.Sprintf(format, args...))
+}
+
+func WithFields(fields map[string]any) *loggerWrapper {
+	return defaultLoggerWrapper.WithFields(fields)
+}
+
+func WithTraceId(ctx context.Context) *loggerWrapper {
```

Comment:
Can we add docstrings for all exported functions?

Reply:
Added.